pkg/controller/statefulset/stateful_pod_control.go

```diff
@@ -18,20 +18,28 @@ package statefulset
 
 import (
 	"fmt"
+	"k8s.io/kubernetes/pkg/features"
+
 	"strings"
 
 	apps "k8s.io/api/apps/v1"
 	"k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
 	errorutils "k8s.io/apimachinery/pkg/util/errors"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	clientset "k8s.io/client-go/kubernetes"
 	appslisters "k8s.io/client-go/listers/apps/v1"
 	corelisters "k8s.io/client-go/listers/core/v1"
 	"k8s.io/client-go/tools/record"
 	"k8s.io/client-go/util/retry"
 )
 
+const (
+	AutoCleanupPersistentVolumeClaim = "storage.alpha.kubernetes.io/auto-clean-pvc"
```

Comment:
We discourage the use of annotations now for alpha features. I think we need to add a new field to the StatefulSet.

Also, since this is a new feature, do we need to go through the KEP process? cc @kow3ns

Reply:
yes, I am very glad to join in this discussion /cc @resouer